### PR TITLE
docs: add log query

### PR DESCRIPTION
**Pull request by zhongzc**
## What's Changed in this PR

*Describe the change in this PR*

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

## Summary by CodeRabbit

- **Documentation**
  - Added a comprehensive guide on using GreptimeDB's query language for log data search and analysis, including full-text search and efficient search configuration.

### CODE REVIEW

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+在 SQL 语句中，可以使用 `MATCHES` 函数来执行全文搜索，这对于日志分析尤其有用。`MATCHES` 函数支持对 `String` 类型的列进行全文搜索。以下是一个典型的使用示例：
+
+```sql
+SELECT * FROM logs WHERE MATCHES(message, 'error OR fail');
+```
+
+`MATCHES` 是一个专门用于全文搜索的函数，它接受两个参数：
+
+- `column_name`：要进行全文搜索的列，该列包含文本数据，列的数据类型必须是 `String`。
+- `search_query`：一个字符串，包含要搜索的关键词和操作符。
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:20]:
```suggestion
- `search_query`：一个字符串，包含要搜索的关键词和操作符，详情请看下文中的[查询语句类型](#查询语句类型)
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+### 简单词项 (Simple Term)
+
+简单的搜索词如下：
+
+```sql
+SELECT * FROM logs WHERE MATCHES(message, 'Barack Obama');
+```
+
+以上查询会被视为 `Barack` 和 `Obama` 两个独立的词项。这意味着查询将匹配包含 `Barack` 或 `Obama` 的所有行。等价于使用 `OR`：
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:32]:
```suggestion
上述 `MATCHES` 中参数 `search_query` 的值 `Barack Obama` 将被视为 `Barack` 和 `Obama` 两个独立的词项，
这意味着该查询将匹配包含 `Barack` 或 `Obama` 的所有行，等价于使用 `OR`：
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+SELECT * FROM logs WHERE MATCHES(message, '"Barack Obama"');
+```
+
+如果需要在短语中包含引号，可以使用反斜杠 `\` 进行转义：
+
+```sql
+SELECT * FROM logs WHERE MATCHES(message, '"He said \"hello\""');
+```
+
+## 全文索引加速全文搜索
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:82]:
```suggestion
## 全文索引加速搜索
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+为了提高全文搜索的效率，特别是在处理大量数据时，建议使用全文索引。全文索引可以显著加快搜索速度，使得即使在庞大的数据集中也能快速定位到包含特定关键词的记录。
+
+### 配置全文索引
+
+在 Pipeline 的配置中，可以指定某列使用全文索引。以下是一个配置示例，其中 `message` 列被设置为全文索引：
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:88]:
```suggestion
在 Pipeline 的配置中，可以[指定某列使用全文索引](./log-pipeline.md#index-字段)。以下是一个配置示例，其中 `message` 列被设置为全文索引：
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    index: fulltext
+  - field: time
+    type: time
+    index: timestamp
+```
+
+### 查看表结构
+
+在数据写入后，可以通过 SQL 命令查看表结构，确认 `message` 列已经被设置为全文索引：
+
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:110]: 使用 DESC 命令会不会更好一些？

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+如果需要在短语中包含引号，可以使用反斜杠 `\` 进行转义：
+
+```sql
+SELECT * FROM logs WHERE MATCHES(message, '"He said \"hello\""');
+```
+
+## 全文索引加速全文搜索
+
+为了提高全文搜索的效率，特别是在处理大量数据时，建议使用全文索引。全文索引可以显著加快搜索速度，使得即使在庞大的数据集中也能快速定位到包含特定关键词的记录。
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:84]:
```suggestion
全文索引是全文搜索的关键配置，尤其是在需要处理大量数据的搜索查询场景中。没有全文索引，搜索操作可能会非常缓慢，影响整体的查询性能和用户体验。通过在 Pipeline 中配置全文索引，可以确保搜索操作能够高效执行，即使是在数据量极大的情况下也能保持良好的性能。
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+ENGINE=mito
+WITH(
+  append_mode = 'true'
+)
+```
+
+### 全文索引的重要性
+
+全文索引是全文搜索的关键配置，尤其是在需要处理大量数据的搜索查询场景中。没有全文索引，搜索操作可能会非常缓慢，影响整体的查询性能和用户体验。通过在 Pipeline 中配置全文索引，可以确保搜索操作能够高效执行，即使是在数据量极大的情况下也能保持良好的性能。
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:131]: 和上面的简介有点重复，且这部分应该放到上面去讲，所以建议挪到上面去

```suggestion
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+## 结语
+
+通过上述介绍，您应该可以更有效地使用 GreptimeDB 的查询功能来搜索和分析日志数据。继续探索更多高级查询功能，以充分利用 GreptimeDB 提供的强大工具。
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:135]: 看起来没有 next step 再进行深入的文档了，也不是很有必要反复强调强大😂，所以建议去掉。

```suggestion
```

**Review by nicecui — COMMENTED** [review]
LGTM 👍

File: docs/nightly/zh/user-guide/log/log-query.md (hunk 5 above)
Comment (zhongzc) [reply docs/nightly/zh/user-guide/log/log-query.md:110]: 暂时不支持用 DESC 展示全文索引

File: docs/nightly/en/user-guide/log/log-query.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+SELECT * FROM logs WHERE MATCHES(message, '"He said \"hello\""');
+```
+
+## Full-Text Index for Accelerated Search
+
+A full-text index is essential for full-text search, especially when dealing with large datasets. Without a full-text index, the search operation could be very slow, impacting the overall query performance and user experience. By configuring a full-text index within the Pipeline, you can ensure that search operations are performed efficiently, even with significant data volumes.
+
+### Configuring Full-Text Index
+
+In the Pipeline configuration, you can [specify a column to use a full-text index](./log-pipeline.md#index-field). Below is a configuration example where the `message` column is set with a full-text index:
```
Comment (nicecui) [inline docs/nightly/en/user-guide/log/log-query.md:88]:
```suggestion
In the Pipeline configuration, you can specify a column to use a full-text index. Below is a configuration example where the `message` column is set with a full-text index:

<!-- In the Pipeline configuration, you can [specify a column to use a full-text index](./log-pipeline.md#index-field). Below is a configuration example where the `message` column is set with a full-text index: -->
```

File: docs/nightly/zh/user-guide/log/log-query.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+SELECT * FROM logs WHERE MATCHES(message, '"He said \"hello\""');
+```
+
+## 全文索引加速搜索
+
+全文索引是全文搜索的关键配置，尤其是在需要处理大量数据的搜索查询场景中。没有全文索引，搜索操作可能会非常缓慢，影响整体的查询性能和用户体验。通过在 Pipeline 中配置全文索引，可以确保搜索操作能够高效执行，即使是在数据量极大的情况下也能保持良好的性能。
+
+### 配置全文索引
+
+在 Pipeline 的配置中，可以[指定某列使用全文索引](./log-pipeline.md#index-字段)。以下是一个配置示例，其中 `message` 列被设置为全文索引：
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/log/log-query.md:88]:
```suggestion
在 Pipeline 的配置中，可以指定某列使用全文索引。以下是一个配置示例，其中 `message` 列被设置为全文索引：

<!-- 在 Pipeline 的配置中，可以[指定某列使用全文索引](./log-pipeline.md#index-字段)。以下是一个配置示例，其中 `message` 列被设置为全文索引： -->
```

**Review by nicecui — COMMENTED** [review]
LGTM

PS：由于 pipeline 的 PR 还没有合并，为了避免 CI 报错，先去掉指向 pipeline 的链接

**Review by nicecui — APPROVED** [review]

**Review by paomian — APPROVED** [review]

**Pull request merged**